dashboard/app/models/lesson.rb

```diff
@@ -243,7 +243,7 @@ def lesson_plan_base_url
     CDO.code_org_url "/curriculum/#{script.name}/#{relative_position}"
   end
 
-  def summarize(include_bonus_levels = false, for_edit: false)
+  def summarize(include_bonus_levels = false, user=nil, for_edit: false)
```

Comment:
This is making me a little nervous given the caching bug that Charlie just fixed.  We sometimes cache the result of the summarize calls and they should generally not take a user parameter.  I'm not sure what the best way to deal with this is -- do we add user-specific information anywhere higher up in the stack?  (Same comment for script_level.)

Reply:
I definitely agree that pulling user data down is less than ideal. Clare and I spent a long time looking through the API calls and couldn't find a better way to pull this in. If it's blocking for this update, I'll need to go back to the drawing board.

Reply:
i agree that we should really try to avoid this. see what we did in [this PR](https://github.com/code-dot-org/code-dot-org/pull/39733/files/c75617c8e72098efb2319a78c43f33aff8819f0c..88b85b646abe53c23ab0d7c82e0041660816601d) to keep the user data out of the summary. since this PR is focused on a specific page, we should theoretically only be hitting this method from a single code path, so it _should_ be relatively straightforward to merge the user data into the summarized data higher up in the call stack as James suggested. happy to talk through this if you want.

Reply:
Would be happy to update this as suggested. It should be noted- I spent a day or two trying and failing to get this merged in higher up, so would love your help.

Reply:
Coming back around to add what was discussed offline- avoiding merging user information here will be fixed in future work in order to get this fix up and running quickly. https://codedotorg.atlassian.net/browse/LP-1857